### DISCUSSION TITLE
[BUG] Outlier Sample Detection - Timeout

### DISCUSSION THREAD

**Discussion by Yarden234** [deepchecks/deepchecks#1776] [Q&A] [answered]
Hello and thank you!
I was trying to run the "Data Integrity Suite", but the test of "Outlier Sample Detection" couldn't run because the runtime was 13 seconds however timeout was set to 10 seconds for this.

Outlier Sample Detection - Train Dataset | Aborting check: calculation was projected to finish in 13.963946929344763 seconds, but timeout was configured to 10 seconds
-- | --

I have an option to solve it, but it's not ideal: I can create that same test but when initialize it, set the timeout to let's say 60 seconds and add the test to the suite, but I think that when running a suite, it will be better to have the possibility to add timeout argument. 
Thank you again!

**Comment by noamzbr** [comment]
Hi @Yarden234, welcome to the community(: 
Currently there is an option to control the timeout using the timeout argument documented [here](https://docs.deepchecks.com/stable/api/generated/deepchecks.tabular.checks.data_integrity.OutlierSampleDetection.html#deepchecks.tabular.checks.data_integrity.OutlierSampleDetection).

If you want to use the data_integrity suite "as is", except that you want a longer timeout for the Outlier Sample Detection check, you can just replace the check in the suite with a check instantiated with the required parameter. For example:

```
from deepchecks.tabular.suites import data_integrity

customized_suite = data_integrity()

customized_suite.remove(8)

from deepchecks.tabular.checks import OutlierSampleDetection

customized_suite.add(OutlierSampleDetection(timeout=15))
```

You can read more on editing existing suites in our [guide](https://docs.deepchecks.com/stable/user-guide/general/customizations/examples/plot_create_a_custom_suite.html#modify-an-existing-suite).

Did this answer your question?

**Comment by matanper** [comment, marked as answer]
Another solution is to pass the `timeout` parameter to the suite method and it will pass it down to any check that uses it:
```
from deepchecks.tabular.suites import data_integrity
suite = data_integrity(timeout=60)
```

**Reply by noamzbr** [reply]
This is actually the easier option.
